### PR TITLE
Add contribution guidelines document

### LINKED ISSUE #51: Provide a contribution guideline
# What

Provide a `CONTRIBUTION.md` with details on how to contribute to the project.

### PR DESCRIPTION
**Pull request by GMinucci**
### What does this PR do
Adds documentation about repo contribution guidelines

### How can it be tested
Not applicable

### Task
Resolves #51 

### Checklist:
- [x] I ran `make validation` locally with success
- [x] I have not introduced new bugs
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new errors

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+We recommend to follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification when naming your commits but it's not strictly imposed as we use a squash merge strategy when merging code.
+
+### Branching strategy
+
+We use a simplified git-flow branching model, on which there are the following types of branches:
+
+- **main**: This branch only contains code that has been released. It represents the public stable release. Release tags are added for versioning control in this branch. Only `develop` or `hotfix` can be merged to this branch.
+
+- **develop**: This is an integration branch. His purpose is to have a staging place where changes can be tested thoroughly before being released publicly (merged into the `main` branch). All branches should base from `develop` except for `hotfix`.
```
Comment (gderaco) [inline docs/CONTRIBUTING.md:21]: `His purpose` -> `Its purpose`

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+- **hotfix**: Critical solutions or additions that requires to be released as soon as possible. This type of branch should alwasy base on latest `main` and can be merged directly without passing through `develop`. In some cases it's advised to also merge the `hotfix` to `develop` for parity sake. Due to it's critical nature, `hotfix` branchs needs to **get approval from at least one of the repo admins**.
```
Comment (gderaco) [inline docs/CONTRIBUTING.md:23]: `it's` -> `its`

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+### Merging strategy
+
+After getting all necessary approvals and fixing any requested change, always go for a **squash merge** strategy. This avoid poluting our commit history while keeping it legible for history searches. 
+
+Before merging, check the generated commit message and look for any necessary update. Keep in mind that it will be used when generating the automatic version changelog at each release.
+
+## Creating a Pull Request
+
+When opening a PR, try to make it's title as meaningfull as possible. Try to include the issue number when applicable as well.
```
Comment (gderaco) [inline docs/CONTRIBUTING.md:45]: `it's` -> `its`
`meaningfull` -> `meaningful`

**Review by gderaco — APPROVED** [review]
looks good - just some typos

**Comment by arthurpalves** [comment]
In _Task_, you can specify:
_resolves #51_

**Review by dileepZion — APPROVED** [review]

**Review by rkyaganti — APPROVED** [review]

**Pull request merged**